zebra-chain/src/work/difficulty.rs

```diff
@@ -269,10 +233,79 @@ impl CompactDifficulty {
         Work::try_from(expanded).ok()
     }
 
-    #[cfg(feature = "getblocktemplate-rpcs")]
-    /// Returns the raw inner value.
-    pub fn to_value(&self) -> u32 {
-        self.0
+    /// Return the difficulty bytes in big-endian byte-order.
+    ///
+    /// Zebra displays difficulties in big-endian byte-order,
+    /// following the u256 convention set by Bitcoin and zcashd.
+    pub fn bytes_in_display_order(&self) -> [u8; 4] {
+        self.0.to_be_bytes()
+    }
+
+    /// Convert bytes in big-endian byte-order into a [`CompactDifficulty`].
+    ///
+    /// Zebra displays difficulties in big-endian byte-order,
+    /// following the u256 convention set by Bitcoin and zcashd.
+    pub fn from_bytes_in_display_order(
+        bytes_in_display_order: &[u8; 4],
+    ) -> Result<CompactDifficulty, BoxError> {
+        let internal_byte_order = u32::from_be_bytes(*bytes_in_display_order);
+
+        let difficulty = CompactDifficulty(internal_byte_order);
+
+        if difficulty.to_expanded().is_none() {
+            return Err("invalid difficulty value".into());
+        }
+
+        Ok(difficulty)
+    }
```

Comment:
This is a bit different from our standard hex implementations, because we're serializing a bitcoin 32-bit floating point value. (Not a `f32`, which is IEEE floating point.)

The snapshots should make sure this is in the correct order.

Reply:
Also, unlike hashes, some difficulty values are invalid.